### PR TITLE
Add BLIP2 Example

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

BLIP-2 is a multi-modal model capable of image-captioning task. It is widely used for natural image-captioning but fine-tuning such a model remains a challenge due to the model's size. The largest model being `blip2-flan-t5-xxl` (~24GB). Hence, we should leverage `peft` to offer users the possibility to fine-tune this model at low cost.

This PR adds BLIP2 support for `peft`. Added also an example script 

cc @pacman100

### CODE REVIEW

File: src/peft/peft_model.py [hunk 1]
```diff
@@ -1034,3 +1034,72 @@ def _prefix_tuning_forward(
...
+        ...     "lora_alpha": 32,
+        ...     "lora_dropout": 0.1,
+        ...     "merge_weights": False,
+        ...     "fan_in_fan_out": False,
+        ...     "enable_lora": None,
+        ...     "bias": "none",
+        ... }
+
+        >>> peft_config = get_peft_config(config)
+        >>> model = AutoModelForCausalLM.from_pretrained("Salesforce/blip2-flan-t5-xl")
```
Comment (younesbelkada) [inline src/peft/peft_model.py:1069]:
```suggestion
        >>> model = AutoModelForVision2Seq.from_pretrained("Salesforce/blip2-flan-t5-xl")
```

File: src/peft/mapping.py [hunk 2]
```diff
@@ -134,9 +137,12 @@ def get_peft_model(model, peft_config):
         model ([`transformers.PreTrainedModel`]): Model to be wrapped.
         peft_config ([`PeftConfig`]): Configuration object containing the parameters of the Peft model.
     """
-
     model_config = model.config.to_dict()
     peft_config.base_model_name_or_path = model.__dict__.get("name_or_path", None)
+
+    if peft_config.task_type == "VISION_2_SEQ" and not isinstance(peft_config, LoraConfig):
+        raise ValueError("Vision2Seq task type is only supported with LORA")
+
```
Comment (pacman100) [inline src/peft/mapping.py:145]: This isn't required if the task type is left unspecified. For unspecified tasks, line 146-148 already use LoRA via `PeftModel` as task-specific sub-class isn't required for LoRA method.

File: src/peft/peft_model.py [hunk 3]
```diff
@@ -1034,3 +1034,72 @@ def _prefix_tuning_forward(
...
+            pixel_values=pixel_values,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+            labels=labels,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            **kwargs,
+        )
```
Comment (pacman100) [inline src/peft/peft_model.py:1105]: Following previous comment, this isn't required if we aren't supporting methods apart from LoRA

File: examples/int8_training/fine_tune_blip2_int8.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+from peft import LoraConfig, get_peft_model
+
+
+# Let's define the LoraConfig
+config = LoraConfig(
+    r=16,
+    lora_alpha=32,
+    lora_dropout=0.05,
+    bias="none",
+    task_type="VISION_2_SEQ",
```
Comment (pacman100) [inline examples/int8_training/fine_tune_blip2_int8.py:29]: Keeping this unspecified will automatically use the LoRA model via PeftModel object as task-specific class isn't a requirement for LoRA

**Review by pacman100 — APPROVED** [review]
Thank you @younesbelkada for iterating, LGTM! 🤗

**Pull request merged**